src/components/settings-menu/NotificationsSection.tsx

```diff
@@ -1,122 +1,401 @@
+  const allTopicsEnabled = Object.values(notifications.topics).every(
+    topic => topic
+  );
+  const allTopicsDisabled =
+    groupOff || Object.values(notifications.topics).every(topic => !topic);
+  const enabledTopics = Object.keys(notifications.topics).filter(
+    // @ts-expect-error: why are you yelling
+    topic => notifications.topics[topic]
+  );
+
+  const composedLabel = () => {
+    if (allTopicsDisabled) {
+      return lang.t('settings.notifications_section.off');
+    }
+
+    if (notifications.enabled) {
+      if (allTopicsEnabled) {
+        return lang.t('settings.notifications_section.all');
+      }
+
+      if (enabledTopics.length > AMOUNT_OF_TOPICS_TO_DISPLAY) {
+        const limitedTopics = enabledTopics
+          .slice(0, AMOUNT_OF_TOPICS_TO_DISPLAY)
+          .map(topic => lang.t(`settings.notifications_section.${topic}`))
+          .join(', ');
+
+        return `${limitedTopics} ${lang.t(
+          'settings.notifications_section.plus_n_more',
+          {
+            n: enabledTopics.length - AMOUNT_OF_TOPICS_TO_DISPLAY,
+          }
+        )}`;
+      } else {
+        return enabledTopics
+          .map(topic => lang.t(`settings.notifications_section.${topic}`))
+          .join(', ');
+      }
+    } else {
+      return lang.t('settings.notifications_section.off');
+    }
+  };
+
+  return <MenuItem.Label text={composedLabel()} />;
+};
+
+const WalletRow = ({ wallet, groupOff, notMainnet }: WalletRowProps) => {
+  const index = useNavigationState(state => state.index);
   const { navigate } = useNavigation();
+  const { notifications } = useNotificationSettings(wallet.address);
+  const [notificationSettings, setNotificationSettings] = useState(
+    notifications
+  );
 
-  const onPress = useCallback(
-    name => {
+  const cleanedUpLabel = useMemo(
+    () => removeFirstEmojiFromString(wallet.label),
+    [wallet]
+  );
+
+  const displayAddress = useMemo(() => {
+    return abbreviations.address(wallet.address, 4, 6);
+  }, [wallet]);
+
+  const walletName = cleanedUpLabel || displayAddress || '';
+
+  useEffect(() => {
+    const data = getAllNotificationSettingsFromStorage();
+    const updatedSettings = data.find(
+      (x: WalletNotificationSettingsType) => x.address === wallet.address
+    );
+    setNotificationSettings(updatedSettings);
+  }, [index, wallet.address]);
+
+  const navigateToWalletSettings = useCallback(
+    (name, address) => {
       navigate(Routes.WALLET_NOTIFICATIONS_SETTINGS, {
         title: name,
+        address,
       });
     },
     [navigate]
   );
 
   return (
-    <MenuContainer>
-      <Menu>
-        <MenuItem
-          disabled
-          rightComponent={<Switch />}
-          size={52}
-          titleComponent={
-            <MenuItem.Title
-              text={lang.t('settings.notifications_section.my_wallets')}
-              weight="bold"
-            />
-          }
+    <MenuItem
+      disabled={notMainnet}
```

Comment:
we use the word `isTestnet` though the app, maybe is best to maintain that language